src/main/java/org/kainos/ea/resources/WebService.java

```diff
@@ -33,4 +37,25 @@ public Response getJobRoles() throws SQLException, DatabaseConnectionException {
             return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
         }
     }
+
+    @GET
+    @Path("/job-specification/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobSpecification( @PathParam("id") int id ) throws SQLException, DatabaseConnectionException {
+
+        try {
+            JobSpecificationResponse jobSpecification = jobsService.getJobSpecification(id);
+
+            if(jobSpecification.getTitle() == null) {
```

Comment:
can you move this logic to the service and unit test (you'll need to throw an exception which you can catch below)